impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -213,9 +213,21 @@ std::optional<Snapshot> DirectionalGaussianBlurFilterContents::RenderFilter(
 
   Scalar x_scale =
       std::min(1.0, 1.0 / std::ceil(std::log2(transformed_blur_radius_length)));
+
+  // Compute the blur radius in the Y direction by projecting it onto the basis
+  // vectors.
+  // Note that this assumes both passes of the blur have the same sigma and are
```

Comment:
I might be misunderstanding, but the API does allow the x and y direction to have different sigmas like here: https://api.flutter.dev/flutter/dart-ui/ImageFilter/ImageFilter.blur.html.

Reply:
Yes, the framework supports distinct sigmaX/Y values. That said, I would guess the vast majority of blur usages have either both values the same, or one value as zero.

Reply:
Whoops, I've been staring at this code for too long. It might be time to refactor this blur to combine the passes and make these optimizations easier.

Reply:
Realized combining the blurs won't really simplify the optimizations because we need to support blurring in arbitrary directions relative to the texture. The input texture can be an arbitrarily complex Contents that was transformed in interesting ways before it was rendered to a texture.

Also, it would be nice to continue supporting arbitrary single direction blur since we could one day expose this to users who want to do high fidelity motion blurs. :)